feat(STOCK-52)

Blaskowitz14 opened the pull request:
- added sync with user-service
- stocks processing in user service
- manual favorites add
- manual favorited delete
- gitignore upd

Blaskowitz14 committed:
feat(STOCK-52)
- added sync with user-service
- stocks processing in user service
- manual favorites add
- manual favorited delete
- gitignore upd

Blaskowitz14 committed:
feat(STOCK-52)
- added sync with user-service
- stocks processing in user service
- manual favorites add
- manual favorited delete
- gitignore upd

vladdossik commented on src/main/java/lissa/trading/tg/bot/config/RabbitMQConfig.java:
```diff
@@ -13,11 +17,48 @@ public class RabbitMQConfig {
+    public Binding userUpdateBinding(Queue userUpdateQueue, TopicExchange userExchange) {
+        return BindingBuilder.bind(userUpdateQueue)
+                .to(userExchange)
+                .with("*.user-update");
```
в йамл

vladdossik commented on src/main/java/lissa/trading/tg/bot/config/RabbitMQConfig.java:
```diff
@@ -13,11 +17,48 @@ public class RabbitMQConfig {
+    public Binding favoriteStocksBinding(Queue userFavoriteStocksQueue, TopicExchange userExchange) {
+        return BindingBuilder.bind(userFavoriteStocksQueue)
+                .to(userExchange)
+                .with("*.favourite-stocks");
```
в йамл

vladdossik commented on src/main/java/lissa/trading/tg/bot/dto/notification/UserFavoriteStocksUpdateDto.java:
```diff
@@ -0,0 +1,19 @@
+@Builder
+public class UserFavoriteStocksUpdateDto {
+    private UUID externalId;
+    List<FavouriteStock> favoriteStocksEntity;
```
private

vladdossik commented on src/main/java/lissa/trading/tg/bot/service/consumer/UserUpdatesNotificationConsumerImpl.java:
```diff
@@ -0,0 +1,100 @@
+    private String userUpdateRoutingKey;
+
+    @RabbitListener(queues = "${integration.rabbit.user-service.user-update-queue.name}")
+    @Transactional
```
нахуя тут открывать транзакцию ?

vladdossik commented on src/main/java/lissa/trading/tg/bot/service/consumer/UserUpdatesNotificationConsumerImpl.java:
```diff
@@ -0,0 +1,100 @@
+    }
+
+    @RabbitListener(queues = "${integration.rabbit.user-service.favourite-stocks-queue.name}")
+    @Transactional
```
аналогично

Blaskowitz14 committed:
feat(STOCK-52)
- massive bugfixes
- accidentally omitted code
- naming & structure & logic changes

Blaskowitz14 committed:
feat(STOCK-52)
- Fixing conflicts caused by an incorrect automatic merge
- minor changes

Blaskowitz14 committed:
feat(STOCK-52)
- rm logs

Blaskowitz14 committed:
feat(STOCK-52)
- rebasing with i to avoid additional commits
- merge fixes

Blaskowitz14 committed:
feat(STOCK-52)
- removed bot token

Blaskowitz14 committed:
feat(STOCK-52)
- moved user-service token to secret

vladdossik commented on k8s/configmap.yaml:
```diff
@@ -15,7 +15,7 @@ data:
   SPRING_RABBITMQ_PASSWORD: "secure_password" # Пароль нового пользователя
   AUTH_SERVICE_URL: "http://lissa-trading-auth-service:8081"
   USER_SERVICE_URL: "http://lissa-trading-user-service:8080"
-  TINKOFF_API_SERVICE_URL: "http://lissa-trading-tinkoff-api-service:8082"
```
ничего тебя не смущает?
удаляешь урл сервиса и пихаешь токен другого сервиса

Blaskowitz14 replied:
тинькофф больше не используется

vladdossik commented on src/main/java/lissa/trading/tg/bot/bot/TelegramBot.java:
```diff
@@ -264,6 +282,41 @@ private void processNewsTickers(Long chatId, String messageText) {
+        if (!messageText.matches(MessageConstants.TICKERS_LIST_PATTERN)) {
+            sendMessage(chatId, MessageConstants.INVALID_TICKERS_MESSAGE);
+        }
+        List<String> tickers = Arrays.asList(messageText.split(","));
```
эти 2 строчки точно можно вынести в отдельный метод

vladdossik commented on src/main/java/lissa/trading/tg/bot/service/consumer/NotificationContext.java:
```diff
@@ -0,0 +1,20 @@
+
+@Component
+public class NotificationContext {
+    private static final ThreadLocal<Boolean> externalSourceContext = ThreadLocal.withInitial(() -> false);
```
нахуя это надо

Blaskowitz14 replied:
Если честно, хз можно ли так делать,
но это нужно чтоб не кидать увед если обновление пользователя вызвано уведом
было в мыслях еще сделать обертку для методов

vladdossik commented on src/main/java/lissa/trading/tg/bot/service/consumer/UserUpdatesNotificationConsumerImpl.java:
```diff
@@ -0,0 +1,105 @@
+    public void receiveUserUpdateNotification(UserUpdateNotificationDto userUpdateDto,
+                                              @Header("amqp_receivedRoutingKey") String routingKey) {
+        log.info("received user update notification: {}", userUpdateDto);
+        if(routingKey.equals(userUpdateRoutingKey)) {
```
выглядит как хуйня прям какая-то

у тебя изначально должно быть настроено так, что читая в этой очереди ты получаешь только нужные тебе сообщения

потому что сообщения попадают в очередь по определенному routingKey


нужно переделать

Blaskowitz14 committed:
feat(STOCK-52)
- logic, structure and naming changes

vladdossik commented on src/main/resources/application-local.yaml:
```diff
@@ -42,8 +42,29 @@ cache:
+      token: YzNSaGRHbHpkR2xqY3kxcGJuUmxjbTVoYkMxclpYaz0K
+
+  rabbit:
+      exchanges:
```
аналогично криво параметры заданы
(см второй пр)

Blaskowitz14 committed:
feat(STOCK-52)
- yaml refactor

Blaskowitz14 committed:
feat(STOCK-52)
- resolved conflicts

Blaskowitz14 committed:
feat(STOCK-52)
- minor changes
